jupyterhub/oauth/provider.py

```diff
@@ -307,7 +314,19 @@ def save_bearer_token(self, token, request, *args, **kwargs):
             - Resource Owner Password Credentials Grant (might not associate a client)
             - Client Credentials grant
         """
-        app_log.debug("Saving bearer token %s", token)
+        log_token = {}
+        log_token.update(token)
+        scopes = token['scope'].split(' ')
+        # TODO:
+        if scopes != ['identify']:
+            raise ValueError("Only 'identify' scope is supported")
+        # redact sensitive keys in log
```

Comment:
Idea for future: If redaction is going to be done in various places beyond this method, perhaps have a small utility that is called.

Reply:
Good idea! I believe the object being redacted is only used in this method, though.

Reply:
If it's only this method, just leave as is :-)